src/python/pants/backend/python/util_rules/pex_requirements.py

```diff
@@ -429,17 +429,21 @@ def _invalid_user_lockfile_error(
         yield (
             f"- The targets use interpreter constraints (`{user_interpreter_constraints}`) that "
             "are not compatible with those used to generate the lockfile "
```

Comment:
"compatible" is a bit generic: the issue is that one must be a `subset` (`InterpreterConstraints.contains`) of the other, right? I think that leaning into that fact below would be helpful too: "either X must be expanded to contain Y, or Y must be shrunk to be a subset of X".